packages/context/src/resolver.ts

```diff
@@ -100,12 +101,15 @@ function resolve<T>(
         return injection.resolve(ctx, injection, s);
       } else {
         // Default to resolve the value from the context by binding key
-        return ctx.getValueOrPromise(injection.bindingKey, {
-          session: s,
-          // If the `optional` flag is set for the injection, the resolution
-          // will return `undefined` instead of throwing an error
-          optional: injection.metadata && injection.metadata.optional,
-        });
+        return ctx.getValueOrPromise(
+          injection.bindingSelector as BindingAddress<unknown>,
```

Comment:
I noticed we are casting `injection.bindingSelector` in several places. Why is this cast necessary? What guarantees are giving us the confidence that `bindingSelector` is always a BindingAddress, despite the type system thinking otherwise?

Reply:
Please `bindingSelector` is now either `BindingAddress` or `BindingFilter`. For the single-value injection by key, it is casted to `BindingAddress`.

Reply:
I added a TS type guard to remove the cast.